### PR TITLE
[Justice Counts] Add a GH action to trigger playtesting deployment from frontend PRs.

### PR DESCRIPTION
**Pull request by brandon-hills**
## Description of the change

Add a GH action to trigger playtesting deployment from frontend PRs.

Note: We added a similar feature for backend PRs [here](https://github.com/Recidiviz/recidiviz-data/pull/26922). 

We have a [deploy_for_playtesting.sh](https://github.com/Recidiviz/recidiviz-data/blob/main/recidiviz/tools/deploy/justice_counts/deploy_for_playtesting.sh) script for deploying to playtesting. However, this has to be run from the command line so only engineers can do it (e.g. if a PM wants to test our a PR, they have to bug an engineer to do a playtesting deploy). This change enables anyone with access to the PR to deploy to playtesting.

The GH action will be triggered manually from the PR. The user must specify the Service Name (defaults to publisher-web) and the [Playtest URL tag.](https://cloud.google.com/run/docs/rollouts-rollbacks-traffic-migration#tags) (defaults to 'test').
To find the manual trigger, navigate to the Actions tab and then find the "Deploy to JC Playtest" action. You will then see a popup that says "This workflow has a workflow_dispatch event trigger."

image

This is where you can trigger the deployment and provide custom arguments. Once the deployment is complete, the access link can be found in the justice-counts-staging [GCP Cloud Run](https://console.cloud.google.com/run?project=justice-counts-staging) page.

Note: It seems like workflow_dispatch is only available after this PR is submitted (so the test below was run from a PR push-event instead).

If the deployment was triggered, the github bot will post the playtesting link in the PR comments as such:

image

Once the deployment is triggered, the recidiviz helper bot will comment in the PR with a link to the service URL.

Screenshot 2024-01-25 at 1 48 29 PM

## Testing
Successfully deployed a playtest service built from the backend `main` branch and this PR's frontend branch (see circled parts of image below).

Screenshot 2024-01-25 at 1 43 52 PM

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by helperbot-recidiviz** [comment]
Playtest deployment was successfully triggered.

**Comment by helperbot-recidiviz** [comment]
Playtest deployment was successfully triggered. Full deployment usually takes 5 minutes. Your playtest link is https://---publisher-web-b47yvyxs3q-uc.a.run.app/

File: .github/workflows/deploy_to_playtesting.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Deploy to Playtest
+
+on:
+  workflow_dispatch:
+    inputs:
+      service_name:
+        description: "Service name."
```
Comment (mxosman) [inline .github/workflows/deploy_to_playtesting.yml:7]: Tiny nit - does this description need to be updated?

**Review by mxosman — APPROVED** [review]
SUPER COOL, @brandon-hills!! Thank you for the super helpful PR description and for implementing this for the FE PRs as well! This is super exciting - can't wait to try this out!

File: .github/workflows/deploy_to_playtesting.yml (hunk 1 above)
Comment (brandon-hills) [reply .github/workflows/deploy_to_playtesting.yml:7]: I don't think so, but correct me if i'm wrong! This input is for the user to specify whether they want it deployed to publisher or the agency dash

Comment (mxosman) [reply .github/workflows/deploy_to_playtesting.yml:7]: Ohhh - I see it now that it's merged:
Screenshot 2024-01-25 at 4 20 34 PM

Haha - for some reason it looked like a default/boilerplate description just looking at the code and not the UI. Thanks for clarifying, Brandon!

**Review by lilidworkin — COMMENTED** [review]
Fantastic, @brandon-hills !

File: .github/workflows/deploy_to_playtesting.yml (hunk 1 above)
Comment (brandon-hills) [reply .github/workflows/deploy_to_playtesting.yml:7]: Yaa it's non-ideal that workflow_dispatch github actions only show up once a PR is merged! oh well

**Pull request merged**